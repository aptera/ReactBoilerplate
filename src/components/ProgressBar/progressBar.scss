.progress-bar-wrapper {

    margin: 10px 0;

    .progress-bar {
        min-height: 40px;
        width: 100%;
        border: 1px solid #000;

        .progress-bar-progress {
            min-height: 40px;
            background-color: rgba(255, 0, 0, 0.503);
            display: flex;
            align-items: center;
            justify-content: flex-end;
        }

        .percentage {
            font-weight: bold;
            padding-right: 10px;
            text-shadow: 2px 2px 10px rgba(0,0,0,.5);
            &::after{
                content: '%'
            }
        }
    }
}