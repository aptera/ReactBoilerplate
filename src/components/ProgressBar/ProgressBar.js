import React, {Component} from 'react';
import './progressBar.scss';

class ProgressBar extends Component {

    constructor(props){
        super(props);
        this.state = {
            name: props.name,
            maxValue: props.maxValue,
            value: props.value,
            currentPercentage: 0,
            color: props.color
        };
        this.stepper = .001;
        this.targetPercentage = (this.props.value / this.props.maxValue);
    }

    tick() {
        this.stepper += .001;
        let newPercentage = this.state.currentPercentage + this.stepper;

        if(newPercentage >= this.targetPercentage)
            clearInterval(this.interval);

        this.setState({
            currentPercentage: newPercentage > 1 ? 1 : newPercentage
        });
    }

    getRadius() {
        return '15px ' + ((this.state.currentPercentage >= 1) ? '15px 15px' : '0 0') + ' 15px';
    }

    componentDidMount() {
        this.interval = setInterval(this.tick.bind(this), 10);
    }

    componentWillUnmount() {
        clearInterval(this.interval);
    }

    render() {
        return (
            <div className="progress-bar-wrapper">
                <label>Progress &reg; Bar</label>
                <div className="progress-bar">
                    <div className="progress-bar-progress" style={{ backgroundColor: this.props.color, width: this.state.currentPercentage * 100 + '%', borderRadius: this.getRadius() }}>
                        <div className="percentage">{Math.round(this.state.currentPercentage * 100)}</div>
                    </div>
                </div>
            </div>
        );
    }
}

export default ProgressBar;