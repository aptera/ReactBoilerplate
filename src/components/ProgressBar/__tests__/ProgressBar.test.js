import React from 'react';

import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

// import the component to put under test.
import ProgressBar from '../ProgressBar';

describe('the progress bar component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <ProgressBar maxValue={100} value={75} color="red" />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});