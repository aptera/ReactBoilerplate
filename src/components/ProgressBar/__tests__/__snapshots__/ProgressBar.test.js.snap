// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the progress bar component renders correctly 1`] = `
<div
  className="progress-bar-wrapper"
>
  <label>
    Progress Â® Bar
  </label>
  <div
    className="progress-bar"
  >
    <div
      className="progress-bar-progress"
      style={
        Object {
          "backgroundColor": "red",
          "borderRadius": "15px 0 0 15px",
          "width": "0%",
        }
      }
    >
      <div
        className="percentage"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
